include/core/CCompressUtils.h

```diff
@@ -35,57 +35,183 @@ namespace core {
 //! a multi-threaded application it would be best to create
 //! one object for each thread.
 //!
-class CORE_EXPORT CCompressUtils : private CNonCopyable {
+class CORE_EXPORT CCompressUtil : private CNonCopyable {
```

Comment:
If you rename the class I think you should also rename the source file to match (plus the corresponding implementation and unit test files).

Reply:
I thought about this a bit.

My rationale for renaming was "utils" suggests to me a collection of utility functionality, i.e. multiple disparate functions and objects, rather than a single (utility) class.

The reason I didn't rename the file was that we actually have two `CCompressUtil` classes, i.e. the inflator and deflator, and these are the types one would actually use from this header. So this is one of the cases where our naming convention for files breaks down, i.e. there isn't a single natural class after which to name the file. Conversely, breaking these classes out into their own header files seems wrong to me too.

What do you think?

Reply:
Other files that contain collections of classes don't have the initial `C` - `FrequencyPredicates.h` and `ProbabilityAggregators.h` for example.  So maybe this file could be just `CompressUtils.h`?  Having the initial `C` suggests it is named after a class.